system/core/dynamic/HTMLHelper.cfc

```diff
@@ -86,7 +86,7 @@ Description :
 				if( NOT listFindNoCase(event.getValue(name="cbox_assets",private=true),thisAsset) ){
 
 					// Load Asset
-					if( findNoCase(".js", thisAsset) ){
+					if( findNoCase( "js", listLast(thisAsset,".") ) ){
```

Comment:
This won't work if a value in the query string has a period in it.
```site.com/assets/myFile.js?foo=bar.baz```
See my comments in the Box team Slack channel.
cc/ @lmajano